ipaserver/install/ipa_cacert_manage.py

```diff
@@ -526,24 +555,31 @@ def _delete_by_nickname(self, nicknames, options):
                 try:
                     tmpdb.verify_ca_cert_validity(ca_nickname)
                 except ValueError as e:
-                    msg = "Verifying \'%s\' failed. Removing part of the " \
-                          "chain? %s" % (nickname, e)
+                    msg = "Verifying removal of \'%s\' failed. Removing " \
+                          "part of the chain? %s" % (nickname, e)
                     if options.force:
                         print(msg)
                         continue
                     raise admintool.ScriptError(msg)
                 else:
                     logger.debug("Verified %s", ca_nickname)
 
-        for _ca_cert, ca_nickname, _ca_trust_flags in ca_certs:
+        removed = False
+        for ca_cert, ca_nickname, _ca_trust_flags, serial in ca_certs:
             if ca_nickname in nicknames:
-                container_dn = DN(('cn', 'certificates'), ('cn', 'ipa'),
-                                  ('cn', 'etc'), api.env.basedn)
-                dn = DN(('cn', nickname), container_dn)
+                if options.serial and options.serial != serial:
+                    continue
                 logger.debug("Deleting %s", ca_nickname)
-                conn.delete_entry(dn)
+                certstore.delete_ca_cert(conn, api.env.basedn, ca_cert)
+
                 return
 
+        if removed is False:
+            raise admintool.ScriptError(
+                "Certificate with name %s and serial number %s not found"
+                % (ca_nickname, options.serial)
+            )
```

Comment:
I wonder, why is there the `removed` variable and `if removed is False`, you never change the value of this variable, therefore it always evaluates as true can't you just omit the variable and if then?

Reply:
Yeah. I think my original intention was to set removed = True but decided to just return instead. Removed in my local tree.